src/update_engine/mock_system_state.cc

```diff
@@ -18,6 +18,10 @@ MockSystemState::MockSystemState()
       this, &dbus_);
 }
 
+MockSystemState::~MockSystemState() {
+  delete mock_update_attempter_;
```

Comment:
I tried but couldn't get it to compile, will try again and see if you can make sense of the errors.